### DISCUSSION TITLE
Problems getting Thonny to run py5

### DISCUSSION THREAD

**Discussion by sonicgrass** [py5coding/py5generator#458] [General]
I keep getting the following error when running the test code as set out in the video.
Process ended with exit code 3221226356.

I have made sure that I have the right version of Java installed and tried reinstalling Thonny.

Computer is Windows 10 Enterprise

Processor
11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz 1.38 GHz

Your help would be appreciated.

Thank you

**Comment by hx2A** [comment]
Hi @sonicgrass !

Can you provide more information? What video are you referring to? Is it @villares 's domestika class?

> Process ended with exit code 3221226356

Does the Sketch run and animate correctly? Or is this just a funny error message you get when you exit? Do you get the same message if you kill the window with the X button in the corner? What if you hit the ESC key?

Sometimes py5 Sketches to have odd messages when exiting. These are hard to eliminate. If they only happen when exiting and don't interfere with a running Sketch, they can be ignored.

**Reply by sonicgrass** [reply]
Hello @hx2A 

Yes the set up video from @villares's Domestika class. The sketch does not run. The blue circle appears, the window does not appear and then I get the error "Process ended with exit code 3221226356."


The test code:
```
def setup():
    size(980, 980)
   
def draw():
    ellipse(mouse_x, mouse_y, 50, 50)
```

Thank you

**Reply by hx2A** [reply]
Hmmmm, I don't know what is causing this. @villares , do you have any insight?

Can you provide more information about your setup? I assume you have followed the setup steps in the Domestika class?

**Reply by sonicgrass** [reply]
Yes, I followed the class setup instructions
https://abav.lugaralgum.com/como-instalar-py5/index-EN.html

**Environment Variables**
JAVA_HOME=C:\Users\scardwell\AppData\Roaming\Thonny\jdk-17
**Interpreter** 
Local Python 3

**thonny-py5mode**
Version 0.4.6a0

Other than that everything is default

**Reply by hx2A** [reply]
Can you find out what version of py5 you have installed? It is different from the thonny-py5mode package. To get this, go to Tools -> Manage Packages and search for py5. If you don't have the latest version. please try upgrading and see if that fixes this.

**Reply by villares** [reply]
Thank you for opening this issue, @sonicgrass, let's try to make this work out!

As you are on Windows, I'd like to ask if you are using the "pre-configured portable Thonny with py5 from a .zip download",  or have you installed it using a fresh Thonny installer from thonny.org and then installed the py5 package and then the plug-in?

**Reply by sonicgrass** [reply]
Thanks @villares 

I installed Thonny with the or install Thonny with [thonny-4.0.2.exe](https://github.com/thonny/thonny/releases/download/v4.0.2/thonny-4.0.2.exe)

I did the plug-in and I don't remember if I did the package part of the instructions. I opened up Manage packages and got the following error : 

PROBLEM IN THONNY'S BACK-END: Exception while handling 'get_active_distributions' (ModuleNotFoundError: No module named 'pkg_resources').
See Thonny's backend.log for more info.

Process ended with exit code 1.

**Reply by sonicgrass** [reply]
I will uninstall and try reinstalling it, to make sure that I got all of the instructions done correctly. 

Thank you

**Reply by villares** [reply]
Great!

If you start with the installer you mentioned. First install just the py5 library with the package manager (not the plugin manager). Then, in the end, install the plug-in, with the plugin manager. Close and re-launch Thonny and using the py5 menu it should prompt to the JDK install.

Another thing to try is the pre-packaged Thonny+py5 download from dropbox...

**Comment by sonicgrass** [comment]
All right, I used the dropbox and followed the instructions and got it to work. 
Thanks for your help.

**Reply by hx2A** [reply]
Excellent, @sonicgrass , very happy you have it working!
